skills/repo_test.go

```diff
@@ -17,18 +17,80 @@
  *
  */
 
-package skills
+package skills_test
 
 import (
 	. "gopkg.in/check.v1"
+
+	. "github.com/ubuntu-core/snappy/skills"
 )
 
 type RepositorySuite struct {
+	t         Type
 	emptyRepo *Repository
 }
 
-var _ = Suite(&RepositorySuite{})
+var _ = Suite(&RepositorySuite{
+	t: &TestType{
+		TypeName: "type",
+	},
+})
 
 func (s *RepositorySuite) SetUpTest(c *C) {
 	s.emptyRepo = NewRepository()
 }
+
+// Tests for Repository.AddType()
+
+func (s *RepositorySuite) TestAddType(c *C) {
+	// Adding a valid type works
+	err := s.emptyRepo.AddType(s.t)
+	c.Assert(err, IsNil)
+	c.Assert(s.emptyRepo.Type(s.t.Name()), Equals, s.t)
+}
+
+func (s *RepositorySuite) TestAddTypeClash(c *C) {
+	t1 := &TestType{TypeName: "type"}
+	t2 := &TestType{TypeName: "type"}
+	err := s.emptyRepo.AddType(t1)
+	c.Assert(err, IsNil)
+	// Adding a type with the same name as another type is not allowed
+	err = s.emptyRepo.AddType(t2)
+	c.Assert(err, ErrorMatches, "duplicate type name")
+	c.Assert(s.emptyRepo.Type(t1.Name()), Equals, t1)
+}
+
+func (s *RepositorySuite) TestAddTypeInvalidName(c *C) {
+	t := &TestType{TypeName: "bad-name-"}
+	// Adding a type with invalid name is not allowed
+	err := s.emptyRepo.AddType(t)
+	c.Assert(err, ErrorMatches, `invalid skill name: "bad-name-"`)
+	c.Assert(s.emptyRepo.Type(t.Name()), IsNil)
+}
+
+// Tests for Repository.Type()
+
+func (s *RepositorySuite) TestType(c *C) {
+	// Type returns nil when it cannot be found
+	t := s.emptyRepo.Type(s.t.Name())
+	c.Assert(t, IsNil)
+	c.Assert(s.emptyRepo.Type(s.t.Name()), IsNil)
+	err := s.emptyRepo.AddType(s.t)
+	c.Assert(err, IsNil)
+	// Type returns the found type
+	t = s.emptyRepo.Type(s.t.Name())
+	c.Assert(t, Equals, s.t)
+}
+
+func (s *RepositorySuite) TestTypeSearch(c *C) {
+	err := s.emptyRepo.AddType(&TestType{TypeName: "a"})
+	c.Assert(err, IsNil)
+	err = s.emptyRepo.AddType(&TestType{TypeName: "b"})
+	c.Assert(err, IsNil)
+	err = s.emptyRepo.AddType(&TestType{TypeName: "c"})
+	c.Assert(err, IsNil)
+	// Type correctly finds types
```

Comment:
The tests below are missing the "correctly" part. They should probably check the TypeName, at least.